app/containers/ItunesContainer/reducer.js

```diff
@@ -0,0 +1,39 @@
+/*
+ *
+ * ItunesContainer reducer
+ *
+ */
+import produce from 'immer';
+import { createActions } from 'reduxsauce';
+import get from 'lodash/get';
+
+export const initialState = { artistName: null, itunesData: [], itunesError: null };
+
+export const { Types: itunesContainerTypes, Creators: itunesContainerCreators } = createActions({
+  requestGetArtist: ['artistName'],
+  successGetArtist: ['data'],
+  failureGetArtist: ['error'],
+  clearArtist: []
+});
+
+/* eslint-disable default-case, no-param-reassign */
+export const itunesContainerReducer = (state = initialState, action) =>
+  produce(state, (draft) => {
+    switch (action.type) {
+      case itunesContainerTypes.REQUEST_GET_ARTIST:
+        draft.artistName = action.artistName;
+        break;
+      case itunesContainerTypes.CLEAR_ARTIST:
+        return initialState;
+      case itunesContainerTypes.SUCCESS_GET_ARTIST:
+        draft.itunesData = action.data;
+        break;
+      case itunesContainerTypes.FAILURE_GET_ARTIST:
+        draft.itunesError = get(action.error, 'message', 'something_went_wrong');
```

Comment:
```
get(action.error, 'message', translate('something_went_wrong'))
```

also, let's move the above to the saga. The reducer should only be responsible for setting properties. No compute/logic here